alica_test_utility/include/alica/test/TestContext.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+
+#include <engine/AlicaContext.h>
+
+#include "BehaviourTrigger.h"
+
+#include <engine/AlicaEngine.h>
+#include <engine/model/Behaviour.h>
+#include <engine/model/EntryPoint.h>
+#include <engine/model/PlanType.h>
+#include <engine/model/Task.h>
+#include <engine/model/Variable.h>
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <string>
+
+namespace alica::test
+{
+class TestBehaviourCreator;
+class BehaviourTrigger;
+class TestContext : public alica::AlicaContext
+{
+public:
+    TestContext(const std::string& roleSetName, const std::string& masterPlanName, bool stepEngine, const essentials::IdentifierConstPtr agentID = nullptr);
+    /**
+     * This allows to get hold of the AlicaEngine, which is
+     * an inaccessible member variable of the AlicaContext, otherwise.
+     * @return Pointer to the AlicaEngine.
+     */
+    alica::AlicaEngine* getEngine();
```

Comment:
We shouldn't give direct engine access to even unit tests of applications. Giving engine access means that our TestContext is not feature complete and application developers need to delve into how engine work to test applications. That's bad.

Reply:
No test is using the getEngine() thing of the TestContext, except for the already existing ones. I am definitely against having several friend classes declared in the engine code, just because the TestContext is not allowed to have a getEngine() interface. 
Though I can have a look, whether the existing test cases can use some more specific interface of the TestContext, but otherwise I would keep it.

Reply:
I adapted all tests to run without engine getter in TestContext.